src/screens/DiscoveryScreenV2/fragment/DomainFragment.js

```diff
@@ -1,182 +1,279 @@
+              name: item.domain_name,
+              image: item.logo,
+              isunfollowed: !item.following,
+              description: item.short_description || null
             }}
-        />
-    </View>
-
-    const __renderDomainItems = () => {
-        if (isFirstTimeOpen) return [<DiscoveryTitleSeparator text="Suggested Domains" key="domain-title-separator" />].concat(domains.map((item, index) => __renderDiscoveryItem(FROM_FOLLOWED_DOMAIN_INITIAL, "followedDomainDiscovery", { ...item, user_id_follower: item.user_id_follower }, index)))
+          />
+        )}
+        {route.name !== 'Followings' && (
+          <DomainList
+            isDomain={true}
+            onPressBody={() => __handleOnPressDomain(item)}
+            handleSetFollow={() => __handleFollow(from, true, item, index)}
+            handleSetUnFollow={() => __handleFollow(from, false, item, index)}
+            item={{
+              name: item.domain_name,
+              image: item.logo,
+              isunfollowed: !item.following,
+              description: item.short_description || null
+            }}
+          />
+        )}
+      </View>
+    );
+  };
 
-        return (
-            <>
-                {followedDomains.map((item, index) => __renderDiscoveryItem(FROM_FOLLOWED_DOMAIN, "followedDomainDiscovery", item, index))}
+  const __renderDomainItems = () => {
+    const followingDomains = [];
+    const unfollowingDomains = [];
 
-                {unfollowedDomains.length > 0 && followedDomains.length > 0 &&
-                    <View style={styles.unfollowedHeaderContainer}>
-                        <Text style={styles.unfollowedHeaders}>{StringConstant.discoveryMoreDomains}</Text>
-                    </View>}
-                {unfollowedDomains.map((item, index) => __renderDiscoveryItem(FROM_UNFOLLOWED_DOMAIN, "unfollowedDomainDiscovery", item, index))}
-            </>
+    domains.forEach((item) => {
+      if (item.user_id_follower) {
+        followingDomains.push(item);
+      } else {
+        unfollowingDomains.push(item);
+      }
+    });
+    if (isFirstTimeOpen)
+      return [
+        followingDomains.map((item, index) =>
+          __renderDiscoveryItem(
+            FROM_FOLLOWED_DOMAIN_INITIAL,
+            'followedDomainDiscovery',
+            {...item, user_id_follower: item.user_id_follower},
+            index
+          )
         )
-    }
+      ]
+        .concat([
+          route.name !== 'Followings' && (
+            <DiscoveryTitleSeparator text="Suggested Domains" key="domain-title-separator" />
+          )
+        ])
+        .concat(
+          unfollowingDomains.map((item, index) =>
+            __renderDiscoveryItem(
+              FROM_FOLLOWED_DOMAIN_INITIAL,
+              'followedDomainDiscovery',
+              {...item, user_id_follower: item.user_id_follower},
+              index + followingDomains.length
+            )
+          )
+        );
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `__renderDomainItems` function has a complex return statement with multiple conditions and mappings. This could be simplified by breaking it down into smaller functions to improve readability and maintainability.

```diff
+  const renderFollowingDomains = () => {
+    return followedDomains.map((item, index) =>
+      __renderDiscoveryItem(FROM_FOLLOWED_DOMAIN, 'followedDomainDiscovery', item, index)
+    );
+  };

+  const renderUnfollowingDomains = () => {
+    return unfollowedDomains.map((item, index) =>
+      __renderDiscoveryItem(FROM_UNFOLLOWED_DOMAIN, 'unfollowedDomainDiscovery', item, index)
+    );
+  };

  const __renderDomainItems = () => {
    if (isFirstTimeOpen) {
      // existing code...
    }

    return (
      <>
        {renderFollowingDomains()}

        {route.name !== 'Followings' &&
          unfollowedDomains.length > 0 &&
          followedDomains.length > 0 && (
            <View style={styles.unfollowedHeaderContainer}>
              <Text style={styles.unfollowedHeaders}>{StringConstant.discoveryMoreDomains}</Text>
            </View>
          )}
        {route.name !== 'Followings' && renderUnfollowingDomains()}
      </>
    );
  };
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->